janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/JanusGraphTraversalSource.java

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 JanusGraph Authors
```

Comment:
This class is needed only for tests. Move it under `janusgraph-test` module

Reply:
As I mentioned before, I think it is not just for the tests.

Reply:
@9lan9 thanks for this code, this will be a good feature. I have one concern about the `JanusGraphTraversalSource` and wanted to confirm. I'm reading this as syntactic sugar that obfuscates the `addE().property(T.id, id)` pattern? My concern is this will diverge from the driver API because a vanilla TinkerPop driver does not have the concept of a `JanusGraphTraversalSource`. I suppose the remote traversal source could be wrapped but I think it would be better to stick to plain Gremlin and have end users use `addE().property(T.id, id)`. This is the same pattern that DSE Graph uses.

Reply:
@twilmes sorry for the late replay. Currently, we use the same pattern of `addE().property(T.id, id)`. Because of the Tinkerpop driver does not have the concept of the `JanusGraphTraversalSource`, I map the  [graphname_traversal](https://github.com/JanusGraph/janusgraph/blob/ca2115d07d038927be1e6e04c5798809f7422f51/janusgraph-core/src/main/java/org/janusgraph/graphdb/management/JanusGraphManager.java#L268)  to the `JanusGraphTraversalSource`, so this can work well.

Reply:
I don't understand why we need this syntactic sugar either. Since this `addE(final String relationId, final String label)` simply calls `addE(label).property(T.id, relationId)`, why not let users do `addE(label).property(T.id, relationId)` directly, which is also consistent with the way to define custom vertex id?

Reply:
@twilmes, @li-boxuan, seems class  `addE(final String relationId, final String label)` is redundant, also for class `JanusGraphTraversalSource`. So I removed this and also worked well. Thanks for the suggestion.